pkg/streamingpromql/operators/function_over_range_vector.go

```diff
@@ -95,12 +116,16 @@ func (m *FunctionOverRangeVector) NextSeries(ctx context.Context) (types.Instant
 
 		// nolint:errorlint // errors.Is introduces a performance overhead, and NextStepSamples is guaranteed to return exactly EOS, never a wrapped error.
 		if err == types.EOS {
+			if m.SeriesValidationFunc != nil {
+				m.SeriesValidationFunc(data, m.metricNames.GetMetricNameForSeries(m.currentSeriesIndex), m.emitAnnotation)
```

Comment:
if `SeriesValidationFunc` always needs metric names we should use the presence of `SeriesValidationFunc` to determine `needMetricNames`.